### PR TITLE
e2e pre-existing config support fix + code dependency improvements

### PR DESCRIPTION
**Pull request by idandash**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,6 +20,13 @@ jobs:
       - run: export DOCKER_TAG=$(echo $CIRCLE_BRANCH | sed -e 's/\//-/g') && docker pull $DOCKER_IMAGE:$DOCKER_TAG && docker tag $DOCKER_IMAGE:$DOCKER_TAG orbs:$DOCKER_TAG
       - run: cd deploy && npm install && ./deploy-staging.sh
 
+  e2e-testnet:
+    machine: true
+    steps:
+      - checkout
+      - run: $(aws ecr get-login --no-include-email --region us-west-2)
+      - run: export DOCKER_TAG=$(echo $CIRCLE_BRANCH | sed -e 's/\//-/g') && docker pull $DOCKER_IMAGE:$DOCKER_TAG && docker run --rm -ti -e E2E_NO_DEPLOY=true -e E2E_PUBLIC_API_ENDPOINT=$E2E_PUBLIC_API_ENDPOINT $DOCKER_IMAGE:$DOCKER_TAG bash -c "cd e2e && ./build.sh && ./test-from-host.sh"
```
Comment (lbeder) [inline .circleci/config.yml]: @netoneko can you move this to a script in the repo itself? It'll make its maintenance easier and will easily also allow us (after some tweaks in the future) to run local tests against staging.

File: .circleci/config.yml [hunk 2]
```diff
@@ -31,3 +38,9 @@ workflows:
           filters:
             branches:
               only: master
+      - e2e-testnet
+          # requires:
+          #   - deploy
+          # filters:
+          #   branches:
+          #     only: master
```
Comment (lbeder) [inline .circleci/config.yml]: Why is this commented out?

File: .circleci/config.yml [hunk 3]
```diff
@@ -20,6 +20,13 @@ jobs:
       - run: export DOCKER_TAG=$(echo $CIRCLE_BRANCH | sed -e 's/\//-/g') && docker pull $DOCKER_IMAGE:$DOCKER_TAG && docker tag $DOCKER_IMAGE:$DOCKER_TAG orbs:$DOCKER_TAG
       - run: cd deploy && npm install && ./deploy-staging.sh
 
+  e2e-testnet:
```
Comment (lbeder) [inline .circleci/config.yml]: Should it be run serially after the build?

File: e2e/src/orbs-client.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { delay } from "bluebird";
 
-import { Transaction, PublicApiClient } from "./public-api-client";
+import { PublicApiClient , Transaction } from "orbs-interfaces";
```
Comment (lbeder) [inline e2e/src/orbs-client.ts]: You have an extra space before the comma.

File: e2e/src/public-api-client.ts [hunk 5]
```diff
@@ -1,54 +1,15 @@
 import * as path from "path";
 import * as caller from "grpc-caller";
-
-const PROTO_PATH = path.resolve(__dirname, "../../projects/architecture/interfaces");
+import { PublicApiClient , getPathToProto } from "orbs-interfaces";
```
Comment (lbeder) [inline e2e/src/public-api-client.ts]: You have an extra space before the comma.

**Comment by netoneko** [comment]
@lbeder I will address your comments in a separate PR that will be a feature rather than hotfix.

**Pull request merged**